services/workflows-service/src/workflow/cron/ongoing-monitoring.cron.ts

```diff
@@ -0,0 +1,218 @@
+            await this.businessReportService.findMany(
+              {
+                where: {
+                  businessId: business.id,
+                  projectId: business.projectId,
+                },
+                orderBy: {
+                  createdAt: 'desc',
+                },
+                take: 1,
+              },
+              projectIds,
+            )
+          )[0];
+
+          if (!lastReceivedReport) {
+            return this.invokeOngoingAuditReport({
+              business,
+              workflowDefinitionConfig,
+              workflowDefinitionId: workflowDefinition.id,
+              currentProjectId: business.projectId,
+              projectIds: projectIds,
+            });
+          }
+
+          const nextRunDate = lastReceivedReport.createdAt; // might change to business metadata mutation to "invokedAt" and not to "receivedAt"
+          nextRunDate.setDate(nextRunDate.getDate() + intervalInDays);
+
+          if (nextRunDate <= new Date()) {
+            await this.invokeOngoingAuditReport({
+              business,
+              workflowDefinitionConfig,
+              workflowDefinitionId: workflowDefinition.id,
+              currentProjectId: business.projectId,
+              projectIds: projectIds,
+              lastReportId: lastReceivedReport.id,
+            });
+          }
+        }
+      }
+    } catch (error) {
+      this.logger.error(`An error occurred: ${isErrorWithMessage(error) && error.message}`);
+    } finally {
+      await this.prisma.releaseLock(this.lockKey);
+    }
+  }
+
+  private async fetchCustomerProcessConfiguration(customers: TCustomerWithDefinitionsFeatures[]) {
+    const customersWithDefinitionsPromise = customers
+      .filter(customer => {
+        customer.features &&
+          customer.features.find(feature => {
+            return (
+              feature.name === CUSTOMER_FEATURES.ONGOING_AUDIT_REPORT.name &&
+              feature.enabled &&
+              customer.definitionConfigs
+            );
+          });
+      })
+      .map(async customer => {
+        const definitionConfig = this.findDefinitionConfig(customer.definitionConfigs);
+        const projectIds = customer.projects.map((project: Project) => project.id);
+
+        const workflowDefinition = await this.workflowDefinitionService.getLastVersionByName(
+          definitionConfig!.definitionName,
+          projectIds,
+        );
+        return {
+          workflowDefinition: workflowDefinition,
+          projectIds: projectIds,
+          definitionConfig: definitionConfig,
+        };
+      });
+
+    return Promise.all(customersWithDefinitionsPromise);
+  }
+
+  private findDefinitionConfig(
+    definitionConfigs: TOngoingAuditReportDefinitionConfig[] | null | undefined,
+  ) {
+    return (
+      definitionConfigs &&
+      definitionConfigs.find(definition => {
+        definition.definitionName === CUSTOMER_FEATURES.ONGOING_AUDIT_REPORT.definitionName &&
```

Comment:
I dont think this check is needed your in ongoing monitoring cron, so you can get straight away the definition id from the config